internal/graph/resolvers/objectmeta.go

```diff
@@ -48,25 +49,38 @@ func (r *objectMeta) Owners(ctx context.Context, obj *model.ObjectMeta) (model.O
 	}
 
 	owners := make([]model.Owner, 0, len(obj.OwnerReferences))
+	// Collect all concurrently.
+	var (
+		mu sync.Mutex
+		wg sync.WaitGroup
+	)
 	for _, ref := range obj.OwnerReferences {
-		u := &kunstructured.Unstructured{}
-		u.SetAPIVersion(ref.APIVersion)
-		u.SetKind(ref.Kind)
-
-		nn := types.NamespacedName{Namespace: pointer.StringDeref(obj.Namespace, ""), Name: ref.Name}
-		if err := c.Get(ctx, nn, u); err != nil {
-			graphql.AddError(ctx, errors.Wrap(err, errGetOwner))
-			continue
-		}
-
-		kr, err := model.GetKubernetesResource(u)
-		if err != nil {
-			graphql.AddError(ctx, errors.Wrap(err, errModelOwner))
-			continue
-		}
-
-		owners = append(owners, model.Owner{Controller: ref.Controller, Resource: kr})
+		ref := ref // So we don't reference the loop variable.
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			u := &kunstructured.Unstructured{}
+			u.SetAPIVersion(ref.APIVersion)
+			u.SetKind(ref.Kind)
+
+			nn := types.NamespacedName{Namespace: pointer.StringDeref(obj.Namespace, ""), Name: ref.Name}
+			if err := c.Get(ctx, nn, u); err != nil {
+				graphql.AddError(ctx, errors.Wrap(err, errGetOwner))
```

Comment:
Similar question as I have below with some of these errors. Would we benefit from logging the errors at all? I'm mostly wondering if strictly relying on the client to see the error is enough or not 😄.

Reply:
We do have some logging in caching client already, I doubt logging additional errors here will help, at least I haven't seen any new errors in my testing.

Reply:
Fair enough. Just wanted to make sure we were considering it 👍